clang/lib/CodeGen/CodeGenFunction.cpp

```diff
@@ -1652,13 +1652,18 @@ void CodeGenFunction::GenerateCode(GlobalDecl GD, llvm::Function *Fn,
   if (getLangOpts().CUDA && !getLangOpts().CUDAIsDevice &&
       getLangOpts().SYCLIsHost && !FD->hasAttr<CUDAHostAttr>() &&
       FD->hasAttr<CUDADeviceAttr>()) {
```

Comment:
I am not very familiar with CUDA so I apologize if these are very basic questions but what is happening in this code. Is a __host__ function being generated for all CUDA __device__ functions during SYCL host compilation? What is the linkage of this function now?

Reply:
During the host compilation, a `__host__` function is generated only if there is `__device__` function that doesn't have a corresponding `__host__` one. This is required by the SYCL host code which compiles also the content of the lambdas that will form SYCL kernels in case of device compilation.

This function is now marked as `dso_local`.
https://github.com/intel/llvm/blob/cd67cf8566b928cba4fd1456b594db3a5b656180/clang/test/CodeGenSYCL/sycl-cuda-host-device-functions.cu#L56-L60